src/promptflow/promptflow/_core/tracer.py

```diff
@@ -192,6 +197,40 @@ def _create_trace_from_function_call(
     )
 
 
+def get_node_name_from_context():
+    tracer = Tracer.active_instance()
+    if tracer is not None:
+        return tracer._node_name
+    return None
+
+
+def enrich_span_with_trace(span, trace):
```

Comment:
Let's add fail safe check around this method.

Reply:
Sure.